Filter/Basic.php

```diff
@@ -203,13 +203,13 @@ public function getParameters()
 
     /**
      * Get the stream resource being filtered.
-     * Maybe available only during filter calls when the closing parameter is
-     * set to false.
+     * Maybe available only during `static::filter` calls when the closing
+     * parameter is set to false.
      *
      * @return  resource
      */
     public function getStream()
     {
-        return $this->stream;
+        return isset($this->stream) ? $this->stream : null;
```

Comment:
Thanks, interesting example BTW.
```php
<?php

require 'vendor/autoload.php';

$GLOBALS['filter_ref'] = null;

// 1. Filter.
class MyFilter extends Hoa\Stream\Filter\Basic {

    public function filter ( $in, $out, &$consumed, $closing ) {

        $GLOBALS['filter_ref'] = $this; /**A: filtername, params, filter, stream*/

        $iBucket = new Hoa\Stream\Bucket($in);
        $oBucket = new Hoa\Stream\Bucket($out);

        while(false === $iBucket->eob()) {

            $consumed += $iBucket->getLength();

            // 2. To upper case.
            $iBucket->setData(mb_strtoupper($iBucket->getData()));

            $oBucket->append($iBucket);
        }

        unset($iBucket);
        unset($oBucket);

        return self::PASS_ON;
    }
}

// 3. Register filter.
Hoa\Stream\Filter::register('my', 'MyFilter');

// 4. Open a stream.
$file = new Hoa\File\Read(__FILE__);

// 5. Apply the filter.
$f = Hoa\Stream\Filter::append($file, 'my', Hoa\Stream\Filter::READ);

// 6. Use the stream!
echo $file->readAll();

var_dump($GLOBALS['filter_ref']); /**B: filtername, params, filter*/
/**
 * doing `$GLOBALS['filter_ref']->stream` will emit a notice here, but who will do that ?
 */
```

So, it seem there is a little oddness in the `php_user_filter` internal.

According to `ReflectionClass` there are only `filtername` & `params` declared.

(A:) At `php_user_filter::filter` call, `filter` & `stream` properties are set.
![image](https://cloud.githubusercontent.com/assets/5406767/10894587/ddac842e-81af-11e5-8d64-01a25289125c.png)

(B:) At `php_user_filter::filter` end  `stream` property is unset.

It's looks like a very old php API in the way properties appear and disappear x)

To conclude; my patch don't do anythings in your example since `getStream` is not called outside of the `filter` (or `Hoa\Stream\Filter\LateComputed::compute`) context.
However, we are in front of somethings unstable and so avoid error can be useful no ?